### DISCUSSION TITLE
MacOs support

### DISCUSSION THREAD

**Discussion by namezys** [bloombloombloom/Bloom#60] [General]
Hey,

I'm trying to build it on mac os.
Actually, I was able to build everything except 2 couple of things. I will fix it soon.

However, I'm not sure if I will be able to connect because I have only an Arduino board. I would like to run something using my devices to check how it works.
Also, I want to add support of simavr.

Is it interesting?

**Comment by navnavnav** [comment]
Hi @namezys,

I'm glad you've made progress with building Bloom on Mac OS, but I hope you haven't spent too long on this, as it's not something I'm looking to take on ATM.

I've considered supporting Mac OS in the past, but have decided against it for many reasons. Some of which can be found below:

- **Apple make it difficult to run Mac OS in a VM, on Linux**
I cannot maintain support for an OS, if I cannot test Bloom on that OS. I don't own a Mac, and Apple forbids the running of Mac OS VMs on non-Mac hardware. As a result, I cannot test Bloom on Mac OS without buying a Mac, which is something I'm not willing to do. If Apple want open-source Linux developers to support Mac OS, they can start treating us a little better (by not expecting us to shell out thousands of £ for a Mac).

- **The time required to maintain support for Mac OS would come at the Linux community's expense**
Maintaining support for Mac OS is not a trivial undertaking. I spend as much of my free time as is possible on Bloom, for the Linux community. If I now add Mac OS support, the time required to maintain it will have to come from somewhere. And in this case, "somewhere" means the Linux community. Any time I spend on supporting Mac OS will be time that would have been spent on the Linux community. To me, this is unacceptable.

Bloom's source code is licensed under the LGPLv3, so you're welcome to add Mac OS support and maintain that version yourself.

Regarding simavr, I'm not sure if there's much benefit in combining it with Bloom. As I understand, simavr already exposes a GDB server, so users can already use it in a similar way to how they use Bloom. The only benefit I can think of is the Insight window. Is there some other benefit that I'm missing? IMHO I wouldn't bother with something like this unless there is significant demand for it.

**Comment by namezys** [comment]
@navnavnav 
I realised that you don't have macOs. Of course, I don't ask you to do it, you do a great job. Right now I want to add it as "unofficial" patch. I will try to keep this patch up to date. I don't think that it is a good time to apply this patch right now. Maybe later if it goes smooth.
Regarding simavr, Insight window is only one benefit. And it's a great benefit for me. I want to do it, it's doesn't look complicated.
Also, I don't have debug hardware and I can't use Bloom right now. So it's a reason why I want to add simavr.

**Reply by navnavnav** [reply]
Ok, if it's something you need and want to take on, awesome!

Let me know if you have any questions - will be happy to help where I can 👍🏽
